CARDS-2362: New Integrated Care Survey questions

marta- opened the pull request:
* Adding a new question at the beginning of the IC survey, and a second page with optional contact info for additional in-person feedback
* Also updating the email templates since the number of survey questions changed.
* Adding the IC survey to the csv exports (survey questions 1-8 only)
* Adding a new setting to export IC contact information to a separate location. **This new export destination is not yet set up in production.**

marta- committed:
CARDS-2362: New Integrated Care Survey questions

sdumitriu commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/IC.xml:
```diff
@@ -207,7 +230,7 @@
 	</node>
 	<node>
-		<name>ic_2</name>
+		<name>ic_3</name>
```
I think this renumbering will require adjusting the copy answers and clinic dashboard settings, right?

marta- replied:
good catch, thanks

marta- replied:
Done in 62f9d54 .

marta- committed:
CARDS-2362: New Integrated Care Survey questions
Updated copy answers and clinic dashboard settings after renumbering the IC questionnaire questions

marta- committed:
CARDS-2362: New Integrated Care Survey questions

Updated the intro text for the IC survey.

marta- committed:
CARDS-2362: New Integrated Care Survey questions

Added export settings for the IC survey

marta- commented:
Rebased on the latest CARDS-2369.

sashaandjic commented:
![image](https://github.com/data-team-uhn/cards/assets/23243244/1d496c40-85c2-48f2-8252-e0f9f34c9e90)

sashaandjic commented:
Should we add labels next to options `1` and `5`?
![image](https://github.com/data-team-uhn/cards/assets/23243244/78caf3af-9450-459b-8cc7-09427a477780)

marta- commented:
> ![image](https://user-images.githubusercontent.com/23243244/274934045-1d496c40-85c2-48f2-8252-e0f9f34c9e90.png)

Looks like it just needs a rebase, we have max answers defaulting to 1 now.

marta- commented:
> Should we add labels next to options `1` and `5`?

I am not allowed to meddle with the first 8 questions, it took months and lots of approvals to get to this point.

sashaandjic commented:
> > ![image](https://user-images.githubusercontent.com/23243244/274934045-1d496c40-85c2-48f2-8252-e0f9f34c9e90.png)
> 
> Looks like it just needs a rebase, we have max answers defaulting to 1 now.

Shouldn't it be radio bttn?

marta- commented:
> > > ![image](https://user-images.githubusercontent.com/23243244/274934045-1d496c40-85c2-48f2-8252-e0f9f34c9e90.png)
> > 
> > 
> > Looks like it just needs a rebase, we have max answers defaulting to 1 now.
> 
> Shouldn't it be radio bttn?

Yes. It's checkbox because maxAnswers is unspecified so it defaults to 0 (unlimited) on this branch. Once rebased on the latest dev, it eill default to 1 thanks to #1606 , and radio buttons will be displayed instead of checkboxes.

sashaandjic commented:
Small complaint: The block of text may look better if not bolded, and/or if split into two parts.
![image](https://github.com/data-team-uhn/cards/assets/23243244/f2531d3e-0d00-450b-a8ae-79bf2f136fa9)

sdumitriu commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/IC.xml:
```diff
@@ -142,6 +146,25 @@
+			<type>String</type>
+		</property>
+		<property>
+			<name>minAnswers</name>
```
Huh, weird, without `maxAnswers` now this is a checkbox list, I can select both Yes and No. Was it always like this for Booleans?

sdumitriu replied:
And in view mode, it's displayed as `1 0`, even if I fix maxAnswers=1.

sdumitriu replied:
Definitely a regression from the MultipleChoice refactoring.

marta- replied:
Indeed, taking care of it soon.

marta- replied:
Resolved in the base pr #1614 and rebased.

sdumitriu commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/IC.xml:
```diff
@@ -727,4 +750,242 @@
+		<primaryNodeType>cards:Section</primaryNodeType>
+		<property>
+			<name>label</name>
+			<value>##### If you would like to provide further input into your experience, please provide your name or that of your family caregiver, and a way to reach you (phone or email). A member of the quality improvement team will reach out to you for interview. Interviews would be maximum 30 minutes and in appreciation of your time, you would receive compensation.</value>
```
It is not entirely clear that this is very much optional. Personally, I would remake this like:

```
I would like to be contacted by:
( ) Phone call
( ) Text
( ) Email
( ) I do not want to be contacted

Name (conditional on answer one of the first 3)

____

Other conditional sections as they are
```

marta- replied:
Agreed

sashaandjic replied:
> It is not entirely clear that this is very much optional. Personally, I would remake this like:
> 
> ```
> I would like to be contacted by:
> ( ) Phone call
> ( ) Text
> ( ) Email
> ( ) I do not want to be contacted
> 
> Name (conditional on answer one of the first 3)
> 
> ____
> 
> Other conditional sections as they are
> ```

Are we doing this?

marta- replied:
Done in d909a37 .

sdumitriu commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/IC.xml:
```diff
@@ -727,4 +750,242 @@
+					<type>String</type>
+				</property>
+				<property>
+					<name>onlyCountries</name>
```
`onlyCountries` or `defaultCountry`? I think they're not going to call someone with a foreign number.

Also, we didn't notice this when testing the phone PR:

![image](https://github.com/data-team-uhn/cards/assets/88663/9f759cfa-74c7-4839-a889-529bdbba26a2)

marta- replied:
This bug is what actually convinced me to use onlyCountries :)

Indeed they won't be calling outside Canada anyway.

sdumitriu commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/IC.xml:
```diff
@@ -727,4 +750,242 @@
+					<type>String</type>
+				</property>
+				<property>
+					<name>maxAnswers</name>
```
Should this be mandatory, now that they indicated they want to be contacted?

marta- replied:
Yes. I was also considering reordering the questions:
- I want to be contacted by... (the only question initially displayed)
- Phone or Email (if they select Phone, Text or Email in the first question), mandatory
- Name (if they select Phone, Text or Email in the first question), mandatory

Wdyt?

marta- replied:
Done in d909a37 .

sdumitriu commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/IC.xml:
```diff
@@ -727,4 +750,242 @@
+					<type>String</type>
+				</property>
+				<property>
+					<name>maxAnswers</name>
```
Should this be mandatory, now that they indicated they want to be contacted?

sdumitriu commented on prems-resources/feature/src/main/features/feature.json:
Works very well.

sdumitriu commented:
> > > > ![image](https://user-images.githubusercontent.com/23243244/274934045-1d496c40-85c2-48f2-8252-e0f9f34c9e90.png)
> > > 
> > > 
> > > Looks like it just needs a rebase, we have max answers defaulting to 1 now.
> > 
> > 
> > Shouldn't it be radio bttn?
> 
> Yes. It's checkbox because maxAnswers is unspecified so it defaults to 0 (unlimited) on this branch. Once rebased on the latest dev, it will default to 1 thanks to #1606 , and radio buttons will be displayed instead of checkboxes.


In dev it is never displayed as checkboxes, even if maxAnswers is 0.

And in view mode it is displayed as `0` or `1`.

sdumitriu commented:
Rebased on dev.

sashaandjic reviewed (approved):
tested - email formatting; import/export of the IC data

![image](https://github.com/data-team-uhn/cards/assets/23243244/25f75d10-6d57-41ba-b6e8-82b0f94df25f)

sashaandjic commented:
Would phone formatting be doable?

marta- commented:
> Would phone formatting be doable?

Can you elaborate? We're already using the PhoneQuestion with formatting.

sashaandjic commented:
> > Would phone formatting be doable?
> 
> Can you elaborate? We're already using the PhoneQuestion with formatting.

Unlike email, users can enter an invalid number with no warning.

marta- commented:
> > > Would phone formatting be doable?
> > 
> > 
> > Can you elaborate? We're already using the PhoneQuestion with formatting.
> 
> Unlike email, users can enter an invalid number with no warning.

I see. The widget we use doesn't support it out of the box. I'll look into it but no promisses.

marta- committed:
CARDS-2362: New Integrated Care Survey questions

Reorganized the feedback section:
* Reordered the questions: contact mode, phone/email, name
* Added the option "I do not want to be contacted" to contact mode
* Made phone or email and name mandatory if selected

marta- commented:
> > > > Would phone formatting be doable?
> > > 
> > > 
> > > Can you elaborate? We're already using the PhoneQuestion with formatting.
> > 
> > 
> > Unlike email, users can enter an invalid number with no warning.
> 
> I see. The widget we use doesn't support it out of the box. I'll look into it but no promises.

If we address this, it will be in a separate PR.

marta- commented:
> QUESTION: Are we going to do this?
> 
> It is not entirely clear that this is very much optional. Personally, I would remake this like:
> 
> I would like to be contacted by: ( ) Phone call ( ) Text ( ) Email ( ) I do not want to be contacted
> 
> Name (conditional on answer one of the first 3)
> 
> Other conditional sections as they are

The feedback section has been reorganized, please retest.

sashaandjic reviewed (approved):
tested - all good

marta- commented:
Rebased on the latest CARDS-2369, no visible changes.

marta- committed:
CARDS-2362: New Integrated Care Survey questions

* Updated style and wording of the IC feedback section intro
* Updated welcome message for all surveys containing the IC questionnaire to match the IC questionnaire intro
* Updated the intro of invitation and reminder emails for all surveys containing the IC questionnaire

marta- commented:
* Updated style and wording of the IC feedback section intro
* Updated welcome message for all surveys containing the IC questionnaire to match the IC questionnaire intro
* Updated the intro of invitation and reminder emails for all surveys containing the IC questionnaire

Needs to be retested and all affected texts needs to be proof-read to catch typos or any awkward phrasing.

sashaandjic reviewed (approved):
tested

marta- commented on prems-resources/feature/src/main/features/feature.json:
```diff
@@ -339,6 +341,18 @@
       "file.name.format": "{questionnaire}.csv",
       "export.format": "csv"
     },
+    "io.uhndata.cards.scheduledcsvexport.ExportConfig~UHN-IC-Survey-Contact":{
+      "name": "UHN-IC-Survey-Contact",
+      "frequency.in.days": 7,
+      "export.schedule": "0 0 2 ? * SAT *",
+      "questionnaires.to.be.exported": [
+        "/Questionnaires/IC"
+      ],
+      "selectors": ".labels.dataFilter:status=SUBMITTED.questionnaireFilter.questionnaireFilter:include=/Questionnaires/IC/ic_feedback",
+      "save.path": "/ic-export",
+      "file.name.format": "{questionnaire}_{period}_survey_contact.csv",
+      "export.format": "csv"
```
We'll have to disable this for now in production until we set up the shared drive.

sashaandjic commented:
All IC fields (including contact) are in the IC csv export.

sdumitriu committed:
CARDS-2362: New Integrated Care Survey questions
Mark the "I don't want to be contacted" question as noneOfTheAbove

sdumitriu committed:
CARDS-2362: New Integrated Care Survey questions
Allow + in email addresses

sdumitriu commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/IC.xml:
```diff
@@ -732,4 +755,331 @@
+				</property>
+				<property>
+					<name>validationRegexp</name>
+					<value>^([a-zA-Z0-9._%-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,})$</value>
```
`+` is often used in emails. A trick to find out who sold your email address to spammers is to use something like `account+website@provider.com`, e.g. `john.doe+ebay@gmail.com`.

sdumitriu commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/apps/cards/clinics/UHN-IC-EDIP/mailTemplates/InitialInvitation/bodyTemplate.html:
```diff
@@ -17,7 +17,7 @@
     </p>
     <p>
-      The survey consists of three questionnaires containing 31 questions and will take 15-20 minutes to complete.
+      The survey consists of three questionnaires containing 33 questions and will take 15-20 minutes to complete.
```
Why do IC-only and IC+ED grow by 1, and the others grow by 2? Was the count wrong before?

marta- replied:
Good catch, my mistake. They should all grow by 2. Initially there was 1 question added to IC and one to IP, but now they're both in IC.

marta- replied:
The numbers have been corrected in 7b7a106 .

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/apps/cards/clinics/UHN-IC-ED/mailTemplates/InitialInvitation/bodyTemplate.html:
```diff
@@ -17,7 +17,7 @@
     </p>
     <p>
-      The survey consists of two questionnaires containing 18 questions and will take 10-15 minutes to complete.
+      The survey consists of two questionnaires containing 19 questions and will take 10-15 minutes to complete.
```
```suggestion
      The survey consists of two questionnaires containing 20 questions and will take 10-15 minutes to complete.
```

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/apps/cards/clinics/UHN-IC-ED/mailTemplates/InitialInvitation/bodyTemplate.txt:
```diff
@@ -4,16 +4,17 @@ You are receiving this email because of your recent visit at a University
 the care team and answering will not impact future care you receive at UHN.
 
-The survey consists of two questionnaires containing 18 questions and will take
+The survey consists of two questionnaires containing 19 questions and will take
```
```suggestion
The survey consists of two questionnaires containing 20 questions and will take
```

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/apps/cards/clinics/UHN-IC-ED/mailTemplates/ReminderNotification/bodyTemplate.html:
```diff
@@ -17,7 +20,7 @@
     </p>
     <p>
-      The survey consists of two questionnaires containing 18 questions and will take 10-15 minutes to complete.
+      The survey consists of two questionnaires containing 19 questions and will take 10-15 minutes to complete.
```
```suggestion
      The survey consists of two questionnaires containing 20 questions and will take 10-15 minutes to complete.
```

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/apps/cards/clinics/UHN-IC-ED/mailTemplates/ReminderNotification/bodyTemplate.txt:
```diff
@@ -1,20 +1,23 @@
 the care team and answering will not impact future care you receive at UHN.
 
-The survey consists of two questionnaires containing 18 questions and will take
+The survey consists of two questionnaires containing 19 questions and will take
```
```suggestion
The survey consists of two questionnaires containing 20 questions and will take
```

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/apps/cards/clinics/UHN-IC/mailTemplates/InitialInvitation/bodyTemplate.html:
```diff
@@ -16,7 +16,7 @@
     </p>
     <p>
-      The questionnaire contains 7 questions and will take about 5 minutes to complete.
+      The questionnaire contains 8 questions and will take about 5 minutes to complete.
```
```suggestion
      The questionnaire contains 9 questions and will take about 5 minutes to complete.
```

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/apps/cards/clinics/UHN-IC/mailTemplates/InitialInvitation/bodyTemplate.txt:
```diff
@@ -2,17 +2,17 @@ Dear ${first_name} ${last_name},
 the care team and answering will not impact future care you receive at UHN.
 
-The questionnaire contains 7 questions and will take about 5 minutes to
+The questionnaire contains 8 questions and will take about 5 minutes to
```
```suggestion
The questionnaire contains 9 questions and will take about 5 minutes to
```

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/apps/cards/clinics/UHN-IC/mailTemplates/ReminderNotification/bodyTemplate.html:
```diff
@@ -16,7 +20,7 @@
     </p>
     <p>
-      The questionnaire contains 7 questions and will take about 5 minutes to complete.
+      The questionnaire contains 8 questions and will take about 5 minutes to complete.
```
```suggestion
      The questionnaire contains 9 questions and will take about 5 minutes to complete.
```

marta- commented on prems-resources/clinical-data/src/main/resources/SLING-INF/content/apps/cards/clinics/UHN-IC/mailTemplates/ReminderNotification/bodyTemplate.txt:
```diff
@@ -1,18 +1,23 @@
 the care team and answering will not impact future care you receive at UHN.
 
-The questionnaire contains 7 questions and will take about 5 minutes to
+The questionnaire contains 8 questions and will take about 5 minutes to
```
```suggestion
The questionnaire contains 9 questions and will take about 5 minutes to
```

marta- committed:
CARDS-2362: New Integrated Care Survey questions

Corrected the number of survey questions announced in the IC and IC+ED emails
